pkg/prefetch/prefetch.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package prefetch
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/containerd/containerd/log"
+)
+
+type prefetchInfo struct {
+	prefetchMap   map[string]string
+	prefetchMutex sync.Mutex
+}
+
+var Pm prefetchInfo
+
+func (p *prefetchInfo) SetPrefetchFiles(body []byte) error {
+	p.prefetchMutex.Lock()
+	defer p.prefetchMutex.Unlock()
+
+	var prefetchMsg []map[string]string
+	if err := json.Unmarshal(body, &prefetchMsg); err != nil {
+		return err
+	}
+
+	if p.prefetchMap == nil {
+		p.prefetchMap = make(map[string]string)
+	}
+	for _, item := range prefetchMsg {
+		image := item["image"]
+		prefetchfiles := item["prefetch"]
+		p.prefetchMap[image] = prefetchfiles
+	}
+
+	log.L.Infof("received prefetch list from nri plugin: %v ", p.prefetchMap)
+	return nil
+}
+
+func (p *prefetchInfo) GetPrefetchMap() map[string]string {
```

Comment:
How about not cloning the whole prefetch map? Just give an image reference as a key argument for the function like `GetPrefetchInfo(image string) string`